account_analytic_salesperson/models/sale_order.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+
+    _inherit = 'sale.order'
+
+    @api.multi
+    def _create_analytic_account(self, prefix=None):
+        """Propagate sale order salesperson through context key"""
+        for order in self:
+            order = order.with_context(analytic_user_id=order.user_id.id)
+            super(SaleOrder, order)._create_analytic_account(prefix=prefix)
+
+    @api.multi
+    def action_confirm(self):
+        """Ensure analytic account is created after sale order confirmation"""
+        res = super().action_confirm()
+        for order in self:
+            if not order.analytic_account_id:
```

Comment:
@grindtildeath 
can we have instead:
     
       for order in self:
            if not order.analytic_account_id:
                order._create_analytic_account()
        return super().action_confirm()

in your case analytic_account_id will not be propagated with context

Reply:
As seen together changing the override from `action_confirm` to `_action_confirm` fixes the issue. Thanks for the suggestion :+1: